core/src/main/java/org/neo4j/cypher/export/CypherDatabaseSubGraph.java

```diff
@@ -0,0 +1,223 @@
+                                .flatMap(label -> StreamSupport.stream(getConstraints(label).spliterator(), false))
+                        ,
+                        StreamSupport.stream(getAllRelationshipTypesInUse().spliterator(), false)
+                                .flatMap(relType -> StreamSupport.stream(getConstraints(relType).spliterator(), false))
+                ).sorted(comp)
+                .collect( Collectors.toList());
+    }
+
+    @Override
+    public Iterable<ConstraintDefinition> getConstraints(Label label) {
+        return transaction.schema().getConstraints(label);
+    }
+
+    @Override
+    public Iterable<ConstraintDefinition> getConstraints(RelationshipType type) {
+        return transaction.schema().getConstraints(type);
+    }
+
+    @Override
+    public Iterable<IndexDefinition> getIndexes(Label label) {
+        return transaction.schema().getIndexes(label);
+    }
+
+    @Override
+    public Iterable<IndexDefinition> getIndexes(RelationshipType type) {
+        return transaction.schema().getIndexes(type);
+    }
+
+    @Override
+    public Iterable<RelationshipType> getAllRelationshipTypesInUse() {
+        return this.relationshipTypes;
+    }
+
+    @Override
+    public Iterable<Label> getAllLabelsInUse() {
+        return this.labels;
+    }
+
+    @Override
+    public long countsForRelationship(Label start, RelationshipType type, Label end) {
+        // even if `MATCH ()-[r]->(:Label) RETURN count(r)` and `MATCH (:Label)-[r]->() RETURN count(r)` leverage on `RelationshipCountFromCountStore`
+        // given that with this method there is planner overhead, we count entities via the `TokenRead`, because it's much faster
+        // and we fallback to query via count store if transaction is not an InternalTransaction
+        if (transaction instanceof InternalTransaction) {
+            final KernelTransaction kernelTransaction = ((InternalTransaction) transaction).kernelTransaction();
+            final TokenRead tokenRead = kernelTransaction.tokenRead();
+            final int startId = getLabelId(start, tokenRead);
+            final int endId = getLabelId(end, tokenRead);
+            if ( labelNotExists(start, startId) || labelNotExists(end, endId) ) {
+                return 0L;
+            }
+            final int relId = tokenRead.relationshipType(type.name());
+
+            return kernelTransaction.dataRead()
+                    .countsForRelationship(startId, relId, endId);
+        }
+        String startNode = cypherNode(start);
+        String endNode = cypherNode(end);
+        String relationship = String.format("[r:%s]", quote(type.name()));
+        return transaction.execute(String.format("MATCH %s-%s->%s RETURN count(r) AS count", startNode, relationship, endNode))
+                .<Long>columnAs("count")
+                .next();
+    }
+
+    @Override
+    public long countsForNode(Label label) {
```

Comment:
same as above, it returns the real count even if the label is not in the cypher query